htdocs/core/class/conf.class.php

```diff
@@ -481,22 +481,7 @@ public function setValues($db)
 			unset($this->global->PROJECT_USE_SEARCH_TO_SELECT);
 		}
 
-		if (!empty($this->productbatch->enabled))
```

Comment:
Currently, goal of this code is because, batch number management is not compatible with some rules of stock autoincrease/decrease. So we force it the rules when module is enabled whatever is setup.
if you move this as init, forcing will not work. We need to be sure to have this setup when module ProductBatch is on (any other options does not work).

Reply:
I can't unset this const if i want to manage stock movement otherwise. It would be better to not allow checking other conf if it's not handle, but allow to disable const

Reply:
Which option for example ?
This is on purpose that we are not able to unset the conf that are forced when module productbatch is enabled. It is because the application itself is not compatible and bugged if we try to use other value than the one forced when productbatch is enabled.
Solution to allow to set this option is to make application working correctly with batch number and the option set. Currently it is not.